server/server.go

```diff
@@ -66,6 +67,8 @@ func (cs *ChatServer) RunGrpc(ctx context.Context, port string) error {
 	grpcServer := grpc.NewServer()
 	chat.RegisterGatewayConnectorServer(grpcServer, cs)
 
+	wg.Add(1)
+	defer wg.Done()
```

Comment:
this will execute as soon as the go routine is started.  I would add the defer into the routine if the idea is to block further up the chain.